publish/rsync.sh

```diff
@@ -0,0 +1,96 @@
+  echo "   $0 -s sources/site/target/fullSite/ -t mars/snapshots/builds/jbosstools-base_4.3.mars/"
+  echo ""
+
+  echo "To push JBT build + update site folders:"
+  echo "   $0 -s sources/site/target/fullSite      -t mars/snapshots/builds/jbosstools-build-sites.aggregate.site_4.3.mars/B${BUILD_NUMBER}-${BUILD_ID}"
+  echo "   $0 -s sources/site/target/fullSite/repo -t mars/snapshots/updates/core/master"
+  echo ""
+
+  echo "To push JBDS build + update site folders:"
+  echo "   $0 -DESTINATION /qa/services/http/binaries/RHDS -URL http://www.qa.jboss.com/binaries/RHDS           -s sources/results                   -t 9.0/snapshots/builds/devstudio.product_master/"
+  echo "   $0 -DESTINATION devstudio@filemgmt.jboss.org:/www_htdocs/devstudio -URL https://devstudio.redhat.com -s sources/site/target/fullSite/repo -t 9.0/snapshots/updates/core/master/"
+  echo ""
+}
+
+if [[ $# -lt 1 ]]; then usage; fi
+
+# read commandline args
+while [[ "$#" -gt 0 ]]; do
+  case $1 in
+    '-DESTINATION') DESTINATION="$2"; shift 1;; # override for JBDS publishing, eg., /qa/services/http/binaries/RHDS
+    '-URL') URL="$2"; shift 1;; # override for JBDS publishing, eg., http://www.qa.jboss.com/binaries/RHDS
+    '-s') SOURCE_PATH="$2"; shift 1;; # ${WORKSPACE}/sources/site/target/repository/
+    '-t') TARGET_PATH="$2"; shift 1;; # mars/snapshots/builds/<job-name>/<build-number>/, mars/snapshots/updates/core/{4.3.0.Alpha1, master}/
+  esac
+  shift 1
+done
+
+# TODO: make sure we have source zips for all aggregates and JBDS
+
+# TODO: make sure we have MD5 sums for all zip/jar artifacts
+
+# TODO: create BUILD_DESCRIPTION variable (HTML string) in Jenkins log containing:
+# link to log, target platforms used, update site/installers folder, coverage report & jacoco file, buildinfo.json
+
+# build the target_path with sftp to ensure intermediate folders exist
+if [[ ${DESTINATION##*@*:*} == "" ]]; then # user@server, do remote op
+  seg="."; for d in ${TARGET_PATH//\// }; do seg=$seg/$d; echo -e "mkdir ${seg:2}" | sftp $DESTINATION/; done; seg=""
+else
+  mkdir -p $DESTINATION/${TARGET_PATH}
+fi
+
+# copy the source into the target
+rsync -arzq --protocol=28 ${SOURCE_PATH}/* ${TARGET_PATH}/
+
+# for JBT aggregates only: regenerate http://download.jboss.org/jbosstools/builds/nightly/*/*/composite*.xml files for up to 5 builds, cleaning anything older than 5 days old
+if [[ ${JOB_NAME/.aggregate} != ${JOB_NAME} ]]; then
+  # regenerate http://download.jboss.org/jbosstools/builds/nightly/*/*/composite*.xml files for up to 5 builds, cleaning anything older than 5 days old
+  # new approach using publish script from Nexus via jbosstools-build-ci-scripts_4.3.x
+  if [[ -f ${WORKSPACE}/sources/util/cleanup/jbosstools-cleanup.sh ]]; then
+    cd ${WORKSPACE}/sources/util/cleanup
+  elif [[ ! -f jbosstools-cleanup.sh ]]; then 
+    wget -q --no-check-certificate -N https://raw.github.com/jbosstools/jbosstools-build-ci/master/util/cleanup/jbosstools-cleanup.sh
```

Comment:
Why is this still here ? Why would there ever be a case where we have a different lookup ?

Why not just assume the jbosstools-cleanup.sh is just next to this script ? 

That would simplify this script a lot and I would rather see the script fail rather than trying to "fix" itself by grabbing some random master version.